Magic/src/main/java/com/elmakers/mine/bukkit/action/builtin/ChatAction.java

```diff
@@ -0,0 +1,51 @@
+package com.elmakers.mine.bukkit.action.builtin;
+
+import com.elmakers.mine.bukkit.action.BaseSpellAction;
+import com.elmakers.mine.bukkit.api.action.CastContext;
+import com.elmakers.mine.bukkit.api.spell.SpellResult;
+import org.bukkit.ChatColor;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+public class ChatAction extends BaseSpellAction {
+
+    private String message;
+
+    private String translate(String str) {
+        return ChatColor.translateAlternateColorCodes('&', str);
+    }
+
+    @Override
+    public void prepare(CastContext context, ConfigurationSection parameters) {
+        super.prepare(context, parameters);
+
+        this.message = context.getMessage(parameters.getString("message"), translate(parameters.getString("message")));
+
+        List<String> l = parameters.getStringList("randomized_messages");
+
+        if (!l.isEmpty()) this.message = translate(l.get(context.getRandom().nextInt(l.size())));
```

Comment:
It's probably more sensible to move the randomization to the perform method.

Reply:
Why?

Reply:
Also, I can't reply to the above two for some reason.

I've used this Action a lot from my MagicAddons project. It's useful for making Babbling Spells and the like, especially when designing Harry Potter spells.

I'm working remotely on a connection that blocks almost all ports without a maven installation so I'll do it when I get to my desktop

Reply:
> Why?
As it is more of a "performing the action" thing rather than "loading the config".
I don't this makes a big difference rn, but we might change the semantics slightly in the future, and then it would only ever print the same message rather than randomizing.

> I've used this Action a lot from my MagicAddons project. It's useful for making Babbling Spells and the like, especially when designing Harry Potter spells.
Ah ok, seems like a reasonable use case.
(Might be a bit spammy, but I guess we can just leave it up to the makers of the spells)

> I'm working remotely on a connection that blocks almost all ports without a maven installation so I'll do it when I get to my desktop
Apart from downloading dependencies via http(s), you shouldn't need any other connections.

Reply:
I'm saying I don't have a Maven installation and a ton of ports and sites are blocked on this network so I'll have to do that when I'm home.